File: frontend/app/models/project.ts
import { z } from "zod";
import { ServerResponse, ServerResponseSchema } from "./default";

export const ProjectSchema = z.object({
  id: z.string(),
  name: z.string(),
  owner_id: z.string(),
  description: z.string().optional(),
  slug: z.string(),
  created_at: z.string(),
  updated_at: z.string(),
  language: z.string().optional(),
  framework: z.string().optional(),
  port: z.string().optional(),
  server_url: z.string().optional(),
  model: z.string().optional(),
});

export type Project = z.infer<typeof ProjectSchema>;

export const GetProjectSchema = ServerResponseSchema.extend({
  data: ProjectSchema,
});

export type GetProject = z.infer<typeof GetProjectSchema>;

export const ProjectsSchema = z.array(ProjectSchema);

export type Projects = z.infer<typeof ProjectsSchema>;

export const GetProjectsSchema = ServerResponseSchema.extend({
  data: ProjectsSchema,
});

export type GetProjects = z.infer<typeof GetProjectsSchema>;

export const CreateProjectFormSchema = z.object({
  prompt: z.string(),
  model: z.string().optional(),
  framework_id: z.string().optional(),
});

export type CreateProjectForm = z.infer<typeof CreateProjectFormSchema>;

export const CreateProjectResponseSchema = ServerResponseSchema.extend({
  data: ProjectSchema,
});

export type CreateProjectResponse = z.infer<typeof CreateProjectResponseSchema>;

File: backend/database/models/project.go
package models

import (
	"time"

	"github.com/guregu/null"
)

type Project struct {
	ID          string      `json:"id" db:"id"`
	OwnerID     string      `json:"owner_id" db:"owner_id"`
	Name        string      `json:"name" db:"name"`
	Slug        string      `json:"slug" db:"slug"`
	Description null.String `json:"description" db:"description"`
	Model       null.String `json:"model" db:"model"`
	ContainerID null.String `json:"-" db:"container_id"`
	Port        null.String `json:"port" db:"port"`
	ServerUrl   null.String `json:"server_url" db:"server_url"`
	Framework   string      `json:"framework" db:"framework"`
	Language    string      `json:"language" db:"language"`
	Metadata    interface{} `json:"metadata" db:"metadata"`
	CreatedAt   time.Time   `json:"created_at,omitempty" db:"created_at,omitempty"`
	UpdatedAt   time.Time   `json:"updated_at,omitempty" db:"updated_at,omitempty"`
	DeletedAt   null.Time   `json:"deleted_at,omitempty" db:"deleted_at"`
}
